### PR TITLE
feat: add account manager class

### PR DESCRIPTION
**Pull request by bbangert**
Because:

* We want to have a single place to manage account related operations in the database layer.

This commit:

* Adds a new AccountManager class and account package.

Closes #FXA-6623

## Checklist

_Put an `x` in the boxes that apply_

- [x] My commit is GPG signed.
- [x] If applicable, I have modified or added tests which pass locally.
- [x] I have added necessary documentation (if appropriate).

### CODE REVIEW

File: libs/shared/db/mysql/account/src/lib/cart.ts [hunk 1]
```diff
@@ -28,16 +27,7 @@ export class Cart extends BaseModel {
...
-    account: {
-      join: {
-        from: 'carts.uid',
-        to: 'accounts.uid',
-      },
-      modelClass: Account,
-      relation: BaseModel.BelongsToOneRelation,
-    },
-  };
+  static relationMappings = {};
```
Comment (bbangert) [inline libs/shared/db/mysql/account/src/lib/cart.ts:31]: Will be dropping the rest of the objection related classes when I refactor the cart manager to use kysley.

File: libs/shared/account/account/src/lib/account.error.ts [hunk 2]
```diff
@@ -0,0 +1,42 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+import { BaseError } from '@fxa/shared/error';
+
+export class AccountError extends BaseError {
+  constructor(message: string, cause?: Error) {
```
Comment (julianpoy) [inline libs/shared/account/account/src/lib/account.error.ts:7]: I'm noticing a pattern where a lot of the libraries are starting to branch out and create their own type signatures for the error constructor. Is this okay, or is this something we want to standardize?
I had all of my errors [here](https://github.com/mozilla/fxa/blob/295130c0b3504ee795032c471181c29dce5fd3cc/libs/shared/contentful/src/lib/errors.ts#L20) use the parent error type signature, if we want to copy that pattern elsewhere.

Comment (bbangert) [reply libs/shared/account/account/src/lib/account.error.ts:7]: We should definitely standardize what we expect this to look like.

File: libs/shared/account/account/src/lib/account.types.ts [hunk 3]
```diff
@@ -0,0 +1,3 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
```
Comment (julianpoy) [inline libs/shared/account/account/src/lib/account.types.ts:3]: [nb] perhaps remove?

File: tsconfig.base.json [hunk 4]
```diff
@@ -27,6 +27,9 @@
       "@fxa/payments/cart": ["libs/payments/cart/src/index.ts"],
       "@fxa/payments/paypal": ["libs/payments/paypal/src/index.ts"],
       "@fxa/shared/contentful": ["libs/shared/contentful/src/index.ts"],
+      "@fxa/shared/account/account": [
+        "libs/shared/account/account/src/index.ts"
```
Comment (julianpoy) [inline tsconfig.base.json:31]: [nb] Not in love with the `account/account` here

**Review by julianpoy — APPROVED** [review]

File: tsconfig.base.json (hunk 4 above)
Comment (bbangert) [reply tsconfig.base.json:31]: Yea, my thought was there would be other account related manager classes, and this one at the data persistence level focuses on the operations for manipulating the Account in our Account system, and there'd be a Devices one, etc. At the moment, it definitely looks..... odd, in isolation.

**Pull request merged**